video/qt_decoder.h

```diff
@@ -418,10 +429,17 @@ class QuickTimeDecoder : public VideoDecoder, public Audio::QuickTimeAudioDecode
 		Graphics::Surface *_projectedPano;
 		Graphics::Surface *_planarProjection;
 
+		// Defining these for the sake of making the swing transition happen
+		// This will be the start point of the transition
+		float _currentFOV = 0;
+		float _currentHFOV = 0;
+		float _currentPanAngle = 0;
+		float _currentTiltAngle = 0;
+
 	private:
 		bool _isPanoConstructed;
-
 		bool _dirty;
+		short _counter = 10;
```

Comment:
Same thing here, now you have a loop which you execute start to finish, so no need to keep it in the object state.